Containerfile

```diff
@@ -1,40 +1,55 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # (c) Copyright 2023 Advanced Micro Devices, Inc.
+# hadolint global ignore=DL3006,DL3059,DL3041,SC1091
+ARG UBI_IMAGE=redhat/ubi9-minimal:latest
+ARG SFPTPD_VERSION
 
-ARG UBI9_VERSION=latest
-
-FROM registry.access.redhat.com/ubi9-minimal:${UBI9_VERSION} AS builder
+FROM $UBI_IMAGE AS builder
+ARG SFPTPD_VERSION
 WORKDIR /src
 
 # git used to derive version number when run from a git clone
-RUN microdnf -y install git tar bzip2 redhat-rpm-config make gcc
+# hadolint ignore=DL3040
+RUN microdnf install -y tar gzip bzip2 redhat-rpm-config make gcc
 
 # Obtain headers from source packages in lieu of libmnl-devel and libcap-devel
-RUN microdnf -y --enablerepo=ubi-9-baseos-source download --source libmnl libcap
+RUN microdnf -y --enablerepo=ubi-*-baseos-source download --source libmnl libcap
 RUN mkdir libcap libmnl
 RUN rpm -i libcap-*.src.rpm \
  && tar xzf ~/rpmbuild/SOURCES/libcap-*.tar.gz --strip-components=1 -C libcap \
  && make -C libcap/libcap install
+# hadolint ignore=DL3003,SC1083
 RUN rpm -i libmnl-*.src.rpm \
  && tar xjf ~/rpmbuild/SOURCES/libmnl-*.tar.bz2 --strip-components=1 -C libmnl \
  && { cd libmnl && ./configure --prefix=/usr && make install; }
 
 # Use build flags from redhat-rpm-config to select the same optimisations
 # and hardening chosen by base platform
-RUN echo \
-  export CFLAGS=\"$(rpm --eval %{__global_cflags})\" \
-  export LDFLAGS=\"$(rpm --eval %{__global_ldflags})\" \
+RUN echo "export \
+  CFLAGS=\"$(rpm --eval '%{__global_cflags}')\" \
+  LDFLAGS=\"$(rpm --eval '%{__global_ldflags}')\"" \
   > /build.env
 
 # Build sfptpd
-COPY . /src/sfptpd
 WORKDIR /src/sfptpd
-RUN make patch_version
-RUN . /build.env && DESTDIR=/staging INST_INITS= make install
-
-FROM registry.access.redhat.com/ubi9-minimal:${UBI9_VERSION} AS runtime
-RUN microdnf install -y libmnl
+RUN --mount=target=. \
+  source /build.env \
+  && make install BUILD_DIR=/src/sfptpd-build DESTDIR=/staging INST_INITS= \
+  && cp LICENSE /staging
+
+FROM $UBI_IMAGE AS runtime
+ARG SFPTPD_VERSION
+LABEL \
+  name="sfptpd" \
+  summary="sfptpd" \
+  description="SF PTPd" \
```

Comment:
Never seen it written like that before! Could we have "Solarflare Enhanced PTP Daemon"?